### ISSUE TITLE
Warnings when login redirect

### ISSUE THREAD

**Issue by RafaelMiquelino** [okta/okta-auth-js#672]
Hi,

I'm getting the following warnings when being redirected from the login page:

```
[okta-auth-sdk] WARN: Use setTokens() instead if you want to add a set of tokens at same time.
It prevents current tab from emitting unnecessary StorageEvent,
which may cause false-positive authState change cross tabs.
````
```
[okta-auth-sdk] WARN: updateAuthState is an asynchronous method with no return, please subscribe to the latest authState update with authStateManager.subscribe(handler) method before calling updateAuthState.
```

This is the piece of code that is throwing the warning:
```javascript
const { OktaAuth } = require('@okta/okta-auth-js');

const siginConfig = require('./signin-config.json');

const authClient = new OktaAuth({
  url: `https://${siginConfig.oktaDomain}`,
  clientId: `${siginConfig.clientId}`,
  redirectUri: window.location.href,
  issuer: `https://${siginConfig.oktaDomain}/oauth2/default`,
});

const getIdToken = function getIdToken(callback) {
  if (authClient.token.isLoginRedirect()) {
    // Parse token from redirect url
    authClient.token.parseFromUrl()
      .then((data) => {
        const { idToken } = data.tokens;
        // Store parsed token in Token Manager
        authClient.tokenManager.add('idToken', idToken);
        callback(idToken);
      });
  } else {
    // Attempt to retrieve ID Token from Token Manager
    authClient.tokenManager.get('idToken')
      .then((idToken) => {
        if (idToken && !authClient.tokenManager.hasExpired(idToken)) {
          callback(idToken);
        } else {
          // You're not logged in, you need a sessionToken
          authClient.token.getWithRedirect({
            responseType: 'id_token',
            scopes: ['openid', 'email', 'profile'],
          });
        }
      });
  }
};
```

The warning is created when I call the function `getIdToken(callback)` and the flow is in a redirect from the login. The version I§m using is the following:
```json
"@okta/okta-auth-js": "^4.0.0"
```

Although the flow is working I'd like to know if there is any means of getting rid of these warnings.

Thanks.

**Comment by shuowu** [comment]
@RafaelMiquelino okta-auth-js implemented `authStateManager` since version 4.1. It basically maintains a synchronous authState in the SDK for the app to get easy access. 

You can check [here](https://github.com/okta/okta-auth-js#authstatemanager) for detailed information.

**Comment by RafaelMiquelino** [comment]
> @RafaelMiquelino okta-auth-js implemented `authStateManager` since version 4.1. It basically maintains a synchronous authState in the SDK for the app to get easy access.
> 
> You can check [here](https://github.com/okta/okta-auth-js#authstatemanager) for detailed information.
> 
> In your case, 1) is suggesting you add tokens to storage at the same to avoid the system triggering multiple authStateChange events when tokens are available to store.
> 2) is suggesting you [subscribe](https://github.com/okta/okta-auth-js#authstatemanagersubscribehandler) to authStateManager to get the latest authState.

Thanks @shuowu.

I managed to get rid of the first one by replacing the line `authClient.tokenManager.add('idToken', idToken)` with `authClient.tokenManager.setTokens(data.tokens)`

However I just manage to get rid of the second one when I subscribe to `authStateManager` but I don't really need that for my use case. Is there any way to just silent this warning?

**Comment by shuowu** [comment]
@RafaelMiquelino You can just subscribe to the authStateManager with a no-op function. It should be able to remove the warning from the console. But we do recommend to track the latest authState (isAuthenticated, tokens) with the AuthStateManager to get sync access to the latest state.

```
oktaAuth.authStateManager.subscribe(() => {});
```
